### PR TITLE
Clean up API package.

### PR DESCRIPTION
**Pull request by Tonkpils**
Use an API Client from the cmd package
Have functions form up their own urls and give them parameters they need

I wanted to get this proof of concept here to get some feedback. Meanwhile, I'll write some tests around this new structure.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -7,16 +7,11 @@ import (
...
 )
 
 var (
-	// UserAgent lets the API know where the call is being made from.
-	// It's set from main() so that we have access to the version.
-	UserAgent string
-
-	UnknownLanguageError = errors.New("the language is unknown")
+	// ErrUnknownLanguage represents an error returned when the language requested does not exist
+	ErrUnknownLanguage = errors.New("the language is unknown")
```
Comment (kytrinyx) [inline api/api.go]: Much more idiomatic!

**Comment by kytrinyx** [comment]
This looks much, much cleaner. I would say go for it.

**Comment by Tonkpils** [comment]
I'll keep adding tests as I go but the main idea is there.

**Comment by lcowell** [comment]
Awesome! Don't forget to give yourself some credit in the changelog.

**Comment by kytrinyx** [comment]
I've added an entry to the changelog for this. Thanks!

**Pull request merged**